library/src/main/java/com/google/android/exoplayer/extractor/mp4/AtomParsers.java

```diff
@@ -1154,4 +1154,119 @@ public AvcCData(List<byte[]> initializationData, int nalUnitLengthFieldLength,
 
   }
 
+  /**
+   * Creates a {@link SampleSizeBox} for the stblAtom passed in.
+   * @param stblAtom the atom to read the sample size atoms from.
+   * @return an implementation of {@link SampleSizeBox} that can be used to parse sample information.
+   * @throws ParserException if no suitable {@link Atom}s are found for parsing.
+     */
+  public static SampleSizeBox sampleSizeBoxFor(Atom.ContainerAtom stblAtom) throws ParserException {
```

Comment:
This could pretty cleanly inlined into the method above as:

```
SampleSizeBox sampleSizeBox;
Atom.LeafAtom stszAtom = stblAtom.getLeafAtomOfType(Atom.TYPE_stsz);
if (stszAtom != null) {
  stszAtom = .....
} else {
  Atom.LeafAtom stz2Atom = stblAtom.getLeafAtomOfType(Atom.TYPE_stz2);
  if (stz2Atom == null) {
    throw new ParserException(....);
  }
  stszAtom = .....
} 
```